extensions/amp-story/1.0/amp-story-reaction-poll-binary.js

```diff
@@ -0,0 +1,172 @@
+ */
+
+import {AmpStoryReaction, ReactionType} from './amp-story-reaction';
+import {CSS} from '../../../build/amp-story-reaction-poll-binary-1.0.css';
+import {createShadowRootWithStyle} from './utils';
+import {dev, devAssert} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {scopedQuerySelector, scopedQuerySelectorAll} from '../../../src/dom';
+import {toArray} from '../../../src/types';
+
+/** @const {string} */
+const TAG = 'amp-story-reaction-poll-binary';
+
+/** @const @enum {number} */
+export const FontSize = {
+  EMOJI: 28,
+  SINGLE_LINE: 16,
+  DOUBLE_LINE: 14,
+};
+
+/**
+ * Generates the template for the binary poll.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildBinaryPollTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-reaction-poll-binary-container"></div>
+  `;
+};
+
+/**
+ * Generates the template for the binary poll option.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildOptionTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-reaction-option">
+      <span class="i-amphtml-story-reaction-option-title"><span></span></span>
+      <span class="i-amphtml-story-reaction-option-percentage-text">0%</span>
+    </div>
+  `;
+};
+
+export class AmpStoryReactionPollBinary extends AmpStoryReaction {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element, ReactionType.POLL, /* bounds */ [2, 2]);
+  }
+
+  /** @override */
+  buildCallback() {
+    super.buildCallback();
+    createShadowRootWithStyle(
+      this.element,
+      dev().assertElement(this.rootEl_),
+      CSS
+    );
+  }
+
+  /** @override */
+  buildComponent() {
+    this.rootEl_ = buildBinaryPollTemplate(this.element);
+    this.attachContent_(this.rootEl_);
+    return this.rootEl_;
+  }
+
+  /**
+   * Gets the options and adds them to the element
+   *
+   * @private
+   * @param {Element} root
+   */
+  attachContent_(root) {
+    this.options_.forEach((option) => {
+      root.appendChild(this.generateOption_(option));
+    });
+    this.adaptFontSize_(root);
+  }
+
+  /**
+   * @private
+   * @param {Element} root
+   */
+  adaptFontSize_(root) {
+    this.mutateElement(() => {
+      let largestFontSize = FontSize.EMOJI;
+      toArray(
+        scopedQuerySelectorAll(
+          dev().assertElement(root),
+          '.i-amphtml-story-reaction-option-title > span'
+        )
+      ).forEach((e) => {
+        if (e.textContent.length <= 3 && largestFontSize >= FontSize.EMOJI) {
+          largestFontSize = FontSize.EMOJI;
+        } else if (
+          e./*OK*/ clientHeight <= 30 &&
+          largestFontSize >= FontSize.SINGLE_LINE
+        ) {
+          largestFontSize = FontSize.SINGLE_LINE;
+        } else if (e./*OK*/ clientHeight > 30) {
+          largestFontSize = FontSize.DOUBLE_LINE;
+        }
+      });
```

Comment:
How this works: If both options have length up to 3 chars (so it's an emoji or short text [yes/no]), font size on post-select will stay at 28px. If both options have one line of height, it will change the font-size to 16, otherwise it will change it to 14px. If the options have different categories, it will choose the smallest font size to allow both items to fit